### PR TITLE
Support service discovery servers to resolve service addresses

### PR DESCRIPTION
**Pull request by azzz**
Includes the possibility to use service discovery implementations to resolve service addresses.

### CODE REVIEW

File: examples/consul/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+This example holds a simple Semaphore HTTP implementation with a service reslved by Consul service discovery
+
+## Getting started
+
+To run this example you need to have Go 1.13> installed on your machine.
+
+Prepare the environment:
+
+1. Run consul `consul agent -dev --config-dir=./awesome-dogs/consul.d -ui`
```
Comment (jeroenrinzema) [inline examples/consul/README.md:11]: let's use a docker container instead.

File: pkg/providers/hcl/services.go [hunk 2]
```diff
@@ -37,6 +36,11 @@ func ParseIntermediateService(parent *broker.Context, manifest Service) (*specs.
 		return nil, err
 	}
 
+	resolver := "default"
```
Comment (jeroenrinzema) [inline pkg/providers/hcl/services.go:39]: should we call the resolver default or dns? I think DNS describes the way on how addresses will be resolved a bit more.

**Review by jeroenrinzema — CHANGES_REQUESTED** [review]

File: examples/consul/README.md (hunk 1 above)
Comment (azzz) [reply examples/consul/README.md:11]: I suppose people who use consul know how to run consul. I will better change the doc a bit

File: website/sidebars.js [hunk 3]
```diff
@@ -24,6 +24,14 @@ module.exports = {
...
       ],
     },
+    {
+      type: 'docs',
+      id: 'service_discovery',
+      label: 'Using Service Discovery',
+      items: [
+        "service_discovery.configuration",
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (azzz) [reply website/sidebars.js]: did it

File: pkg/providers/hcl/services.go (hunk 2 above)
Comment (azzz) [reply pkg/providers/hcl/services.go]: probably yes, renamed.

**Review by jeroenrinzema — APPROVED** [review]

**Pull request merged**